src/geop/main.py

```diff
@@ -1,11 +1,60 @@
-from flask import Flask
+import os
+import time
+
+from flask import Flask, request, jsonify
+import numpy as np
+import rasterio
 
 app = Flask(__name__)
 
+DATA_PATH = '/usr/data/'
+
+
+@app.route('/counts', methods=['POST'])
+def count():
+    """
+    Perform a rudimentary cell count analysis on a portion of a
+    provided raster
+
+    Query String Args:
+        filename: file name of a valid geotiff in DATA_DIR
+    """
+
+    raster = request.args.get('filename')
+    if not raster:
+        return handle_error('filename parameter is required')
```

Comment:
Not a big deal, but it looks like Flask error handlers do something pretty similar to this, but with decorators, so potentially a little cleaner:

http://flask.pocoo.org/docs/0.11/patterns/apierrors/